crates/router/src/core/payments/routing.rs

```diff
@@ -254,7 +322,13 @@ pub async fn perform_static_routing_v1<F: Clone>(
             .change_context(errors::RoutingError::ConnectorSelectionFailed)?,
 
         CachedAlgorithm::Advanced(interpreter) => {
-            let backend_input = make_dsl_input(payment_data)?;
+            let backend_input = match operation_data {
+                routing::TransactionData::Payment(payment_data) => make_dsl_input(payment_data)?,
+                #[cfg(feature = "payouts")]
+                routing::TransactionData::Payout(payout_data) => {
+                    make_dsl_input_for_payouts(payout_data)?
+                }
+            };
```

Comment:
Don't rely on `RoutableConnectors` as an output for Payouts. `RoutableConnectors` is for payments only. You can use a custom `PayoutConnectors` enum as the output type for the DSL.

Reply:
Added an issue for this - https://github.com/juspay/hyperswitch/issues/3671

Will raise it as a separate PR.